interpreter/meta/jslib/wast.ml

```diff
@@ -0,0 +1,26 @@
+(* Implements a wrapper library that allows the use of the reference
+ * interpreter's encode/decode functionality in JavaScript.
+ *)
+open Js_of_ocaml
+
+let _ =
+  Js.export "WebAssemblyText"
+    (object%js (_self)
+
+      method encode (s : Js.js_string Js.t) : (Typed_array.arrayBuffer Js.t) =
+        let def = Parse.string_to_module (Js.to_string s) in
+        let bs =
+          match def.Source.it with
+          | Script.Textual m -> (Encode.encode m)
+          | Script.Encoded (_, bs) -> bs
+          | Script.Quoted (_, _) -> failwith "Unsupported" in
+        let buf = new%js Typed_array.arrayBuffer (String.length bs) in
+        let u8arr = new%js Typed_array.uint8Array_fromBuffer buf in
+        String.iteri (fun i c -> Typed_array.set u8arr i (int_of_char c)) bs; buf
+
+      method decode (buf : Typed_array.arrayBuffer Js.t) width : (Js.js_string Js.t) =
```

Comment:
Perhaps by declaring the parameter
```
(optwidth : int Js.optdef)
```
and then doing
```
let width = Js.Optdef.get optwidth (fun () -> 80) in
```
it is possible to emulate the previous default parameter?

Reply:
You can achieve that behavior using JS.Unsafe.* functions.
Maybe with `val decode = Js.Unsafe.callback (fun bud width -> ...)`